### PR TITLE
Add `dnb_investigation_id` field ahead of new investigation API endpoint

### PR DESCRIPTION
**Pull request by currycoder**
### Description of change

This PR adds a new field to the Company model: `dnb_investigation_id`.  This field will store a UUID generated by dnb-service (https://github.com/uktrade/dnb-service/blob/master/company/models.py#L351) when dnb-service's new "company investigation" API endpoint is called.
This PR also adds some TODO comments to indicate that certain existing company investigation logic/fields will be removed in future.

### Background

Blue team is working on a feature to improve the way that new company investigations are created and managed with D&B.  This feature relates to the situation where a user creates a new company on Data Hub and cannot find the company that they are looking for through D&B's search results.  

Currently, a stub company is created and an email notification is triggered to launch an investigation with D&B.  This solution is not scalable and we need to move to a batched process - where investigations are queued up and sent to D&B daily in a CSV spreadsheet.

The new solution involves adding an extra "company investigation" endpoint to dnb-service.  The aspiration is that this endpoint will be called every time we need to launch a new company investigation.  Investigations will be persisted on dnb-service and will be submitted to D&B daily in a batched spreadsheet.

Data Hub's part in this is as follows:
- It will provide a proxy company investigation endpoint which will allow Data Hub frontend to create new investigations relating to an existing Data Hub company (this company might have just been created as a stub during the company creation unhappy path).  This endpoint will transform and proxy calls through to D&B in a very similar way to the D&B change-request endpoint.
- It will record the company investigation ID provided by dnb-service so that investigation responses from D&B can be matched up to a Data Hub company.  Responses from D&B will be handled manually initially, but eventually the aspiration is to handle responses from D&B in a more automated way.
- Once this new approach for creating investigations is in use, we will remove the `dnb_investigation_data` and notify call which relates to the existing way of creating D&B company investigations.

This PR lays the groundwork for the proxy API - which will follow in a subsequent PR.

### Checklist

* [x] If this is a releasable change, has a news fragment been added?

  <details>
  <summary>Explanation</summary>
  
  A news fragment is required for any releasable change (i.e. code that runs in or affects production) so that a corresponding changelog entry is added when releasing.
  
  Check [changelog/README.md](https://github.com/uktrade/data-hub-api/blob/master/changelog/README.md) for instructions.
  
  </details>
  
* [x] Has this branch been rebased on top of the current `develop` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [x] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/develop/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

File: changelog/company/company-dnb-investigation-id-field.db.md [hunk 1]
```diff
@@ -0,0 +1 @@
+A field `dnb_investigation_id` was added to the `company_company` table.
```
Comment (elcct) [inline changelog/company/company-dnb-investigation-id-field.db.md]: Would be good to add information about the type of field.

File: datahub/dnb_api/test/conftest.py [hunk 2]
```diff
@@ -153,3 +153,43 @@ def investigation_payload():
         'sector': Sector.renewable_energy_wind.value.id,
         'uk_region': UKRegion.east_midlands.value.id,
     }
+
+
+@pytest.fixture
+def base_company_dict():
```
Comment (elcct) [inline datahub/dnb_api/test/conftest.py]: I would rename it to either `get_base_company_dict` or `create_base_company_dict`.

**Comment by currycoder** [comment]
Rebasing to change field type from `CharField` to `UUIDField`.

File: datahub/company/models/company.py [hunk 3]
```diff
@@ -231,10 +231,18 @@ class GreatProfileStatus(models.TextChoices):
         on_delete=models.SET_NULL,
         related_name='+',
     )
+    dnb_investigation_id = models.UUIDField(
+        null=True,
```
Comment (reupen) [inline datahub/company/models/company.py]: I think we'll still want `blank=True` if the field is optional (it relates to validation: https://docs.djangoproject.com/en/3.0/ref/models/fields/#blank).

**Pull request merged**